tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/CircuitBreakerExceptionHierarchyTest.java

```diff
@@ -0,0 +1,332 @@
+        }
+        catch (CircuitBreakerOpenException ex) {
+            return CircuitState.OPEN;
+        }
+        catch (Throwable ex) {
+            assertEquals(ex, exception);
+            return CircuitState.CLOSED;
+        }
+    }
+
+    // serviceB: @CircuitBreaker(failOn = {Exception.class, E1.class}, skipOn = {E0.class, E2.class})
+
+    @Test
+    public void serviceBthrowsException() {
+        // serviceB directly mentions Exception in failOn
+        assertEquals(invokeserviceB(new Exception()), CircuitState.OPEN);
+    }
+
+    @Test
+    public void serviceBthrowsE0() {
+        // serviceB directly mentions E0 in skipOn
+        assertEquals(invokeserviceB(new E0()), CircuitState.CLOSED);
+    }
+
+    @Test
+    public void serviceBthrowsE1() {
+        // serviceB directly mentions E1 in failOn, but it also mentions E0 in skipOn and E1 <: E0
+        assertEquals(invokeserviceB(new E1()), CircuitState.CLOSED);
+    }
+
+    @Test
+    public void serviceBthrowsE2() {
+        // serviceB directly mentions E2 in skipOn
+        assertEquals(invokeserviceB(new E2()), CircuitState.CLOSED);
+    }
+
+    @Test
+    public void serviceBthrowsE2S() {
+        // serviceB mentions E2 in skipOn and E2S <: E2
+        assertEquals(invokeserviceB(new E2S()), CircuitState.CLOSED);
+    }
+
+    @Test
+    public void serviceBthrowsE1S() {
+        // serviceB mentions E1 in failOn and E1S <: E1, but it also mentions E0 in skipOn and E1S <: E0
+        assertEquals(invokeserviceB(new E1S()), CircuitState.CLOSED);
+    }
+
+    @Test
+    public void serviceBthrowsE0S() {
+        // serviceB mentions E0 in skipOn and E0S <: E0
+        assertEquals(invokeserviceB(new E0S()), CircuitState.CLOSED);
+    }
+
+    @Test
+    public void serviceBthrowsRuntimeException() {
+        // serviceB mentions Exception in failOn and RuntimeException <: Exception
+        assertEquals(invokeserviceB(new RuntimeException()), CircuitState.OPEN);
+    }
+
+    @Test
+    public void serviceBthrowsError() {
+        // serviceB doesn't mention Error (nor any of its superclasses) neither in failOn nor skipOn
+        assertEquals(invokeServiceA(new Error()), CircuitState.CLOSED);
+    }
+
+    private CircuitState invokeserviceB(Throwable exception) {
+        try {
+            service.serviceB(exception);
+            throw new AssertionError("Exception not thrown from serviceB");
+        }
+        catch (Throwable ex) {
+            assertEquals(ex, exception);
+        }
+        
+        try {
+            service.serviceB(exception);
+            throw new AssertionError("Exception not thrown from serviceB");
+        }
+        catch (CircuitBreakerOpenException ex) {
+            return CircuitState.OPEN;
+        }
+        catch (Throwable ex) {
+            assertEquals(ex, exception);
+            return CircuitState.CLOSED;
+        }
+    }
+
+    // serviceC: @Retry(retryOn = {E1.class, E2.class}, abortOn = E0.class)
```

Comment:
Forgot to replace retry stuff with circuit breaker stuff.

Reply:
Well spotted, thanks. Now fixed.